src/fuzzy.c

```diff
@@ -933,32 +961,36 @@ fuzzymatches_to_strmatches(
     static int
 has_match(char_u *needle, char_u *haystack)
 {
-    while (*needle != NUL)
+    if (!needle || !haystack || !*needle)
+	return !*needle;
+
+    char_u *n_ptr = needle;
+    char_u *h_ptr = haystack;
+
+    while (*n_ptr)
     {
-	int n_char = mb_ptr2char(needle);
-	char_u *p = haystack;
-	int h_char;
-	int matched = FALSE;
+	int n_char = mb_ptr2char(n_ptr);
+	int n_upper = MB_TOUPPER(n_char);
```

Comment:
That seems unneccessary. we can just keep using MB_TOUPPER(n_char) when we need it